diff --git a/SystemUI/src/com/android/systemui/statusbar/SignalClusterView.java b/SystemUI/src/com/android/systemui/statusbar/SignalClusterView.java
index 822303c..9e9e3a1 100644
--- a/SystemUI/src/com/android/systemui/statusbar/SignalClusterView.java
+++ b/SystemUI/src/com/android/systemui/statusbar/SignalClusterView.java
@@ -41,6 +41,7 @@ import android.widget.TextView;
 
 import com.android.systemui.R;
 import com.android.systemui.statusbar.phone.StatusBarIconController;
+import com.android.systemui.statusbar.phone.StatusBarIconController.MaxChildCountChangListener;
 import com.android.systemui.statusbar.policy.NetworkController.IconState;
 import com.android.systemui.statusbar.policy.NetworkControllerImpl;
 import com.android.systemui.statusbar.policy.SecurityController;
@@ -108,6 +109,8 @@ public class SignalClusterView
     private boolean mBlockEthernet;
     private boolean mBlockNetworkLabel;// added by yangfan
     private Handler mDelaySignaHandler = new  Handler();
+    
+    private MaxChildCountChangListener mMaxChildCountChangListener;
 
     public SignalClusterView(Context context) {
         this(context, null);
@@ -888,5 +891,30 @@ public class SignalClusterView
             state.setLabelVisibility(enable, noServiceEnable);
         }
     }
+
+	public void setMaxChildCountChangListener(MaxChildCountChangListener maxChildCountChangListener) {
+		this.mMaxChildCountChangListener = maxChildCountChangListener;
+	}
+	
+	// 获取子view 个数
+	@Override
+	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+		super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+		int childCount = getChildCount();
+		int notGoneChild = 0;
+		View v ;
+		for (int i = 0; i < childCount; i++) {
+			v = getChildAt(i);
+			if (null != v && v.getVisibility() == VISIBLE && v.getMeasuredWidth() > 3) {//1dp是 space view
+				notGoneChild ++;
+				Log.e(TAG, "notGoneChild : " + v.getClass().getSimpleName() + " , width : " +  v.getMeasuredWidth() );
+			}
+		}
+		if (null != mMaxChildCountChangListener) {
+			Log.e(TAG, "notGoneChild : " + notGoneChild);
+			mMaxChildCountChangListener.onMaxChildCountChanged(notGoneChild >= 5);
+		}
+	}
+    
 //added by yangfan end 
 }
