diff --git a/SystemUI/src/com/android/systemui/statusbar/phone/StatusBarIconController.java b/SystemUI/src/com/android/systemui/statusbar/phone/StatusBarIconController.java
index c28fdb4..cb95948 100644
--- a/SystemUI/src/com/android/systemui/statusbar/phone/StatusBarIconController.java
+++ b/SystemUI/src/com/android/systemui/statusbar/phone/StatusBarIconController.java
@@ -119,6 +119,7 @@ public class StatusBarIconController implements Tunable {
         mSignalCluster = (SignalClusterView) statusBar.findViewById(R.id.signal_cluster);
         mNotificationIconArea = statusBar.findViewById(R.id.notification_icon_area_inner);
         mNotificationIcons = (IconMerger) statusBar.findViewById(R.id.notificationIcons);
+        mSignalCluster.setMaxChildCountChangListener(mNotificationIcons);//监听信号图标最大个数
         mMoreIcon = (ImageView) statusBar.findViewById(R.id.moreIcon);
         mNotificationIcons.setOverflowIndicator(mMoreIcon);
         mStatusIconsKeyguard = (LinearLayout) keyguardStatusBar.findViewById(R.id.statusIcons);
@@ -566,5 +567,13 @@ public class StatusBarIconController implements Tunable {
     public void setNotificationAreaVisibilty(int vis){
         mLeftClockLayout.setVisibility(vis);
     }
+    
+    /**
+     * 监听 信号图标 + 通知图标 是否达到7个
+     * @author mare
+     */
+    public static interface MaxChildCountChangListener{
+    	public void onMaxChildCountChanged(boolean exceed);
+    }
 	//added  by yangfan
 }
